### ISSUE TITLE
As a beginner N4JS user, I miss the following features

### ISSUE THREAD

**Issue by qtran-n4** [eclipse/n4js#1301]
During an introduction of n4js to a developer new to n4js, we have discovered several missing features/suggestions for improvement:

1. ES6 imports -> **DONE** (see PRs #1309, #1315)
The latest master's version emits System.js code which cannot be understood by Webpack. We are currently in the process of changing N4JS to emit ES6 imports in the transpiled code (See https://github.com/eclipse/n4js/issues/1281). Once this is done, Webpack can natively understand the transpiled code again. 

2. Import paths in transpiled code should be relative paths -> **DONE** (see PR #1309)
Currently, the n4js compiler uses absolute paths for import statements. This causes both node.js and Webpack to not be able to find imported modules correctly. 
Once https://github.com/eclipse/n4js/issues/1281 is done, relative paths of imports will be emitted in transpiled code.

3. Spread operator for copying object is missing
Especially in JavaScript applications with Redux, the spread operator is often used for copying object literals, e.g.

```
return { ...state, visibilityFilter: action.filter }
```
This spread operator is currently missing in n4js.

4. Array<>.fill is missing
The new ES6 Array<>.fill is missing in `n4js-runtime-ecma2015.n4jsd` file

4. In-depth n4js tutorial is out of date
The in-depth tutorial https://www.eclipse.org/n4js/userguides/tutorial.html#tutorial is out of date

5. Guides/tutorials for developing web apps (e.g. using React) in N4JS are missing -> **DONE** (see [chessgame](https://www.eclipse.org/n4js/userguides/n4js-tutorial-chess/n4js-tutorial-chess.html))
In addition to SDKs, N4JS is currently being used a lot for developing web apps. Tutorials/guides for developing web apps in N4JS are missing. During the introduction above, I have created some tutorials/examples showing how to use N4JS in combination with React/Redux:

* Develop web apps with N4JS and React: https://github.com/qtran-n4/tictactoe
* Develop web apps with N4JS, React and Redux: https://github.com/qtran-n4/tictactoe-redux

6. Installing npm package (via Quickfix) does not always index it
We have observed the following situation: Some external npm dependency is missing, we use Quickfix ( e.g. run npm/yarn in this project) to install the missing dependency. Afterwards, the library are sometimes not recognized by the IDE. Running Library Manager's `Re-build node_modules` is then needed to resolve the issue.

7. N4JS spec is not that helpful to someone who just wants to start using N4JS right away
This is a real chat message:
```
hi quang, i checked the specs of n4js but could not find any detail. Could you please explain what is the + notation ?
it is like Object+
```

**Comment by halitogunc** [comment]
- Quick fix feature is missing.

```
function foo(a) {
  return a;
}

const x: number = 5;
foo(x);
```

![image](https://user-images.githubusercontent.com/13641726/56200419-0e432b80-603f-11e9-85d0-cec762f3df9c.png)

- On the other hand, it would be helpful to see type of variable which assigned to the function that returns value with types like below.

![image](https://user-images.githubusercontent.com/13641726/56201065-5d3d9080-6040-11e9-9cb7-2fd25e704147.png)

**Comment by mmews-n4** [comment]
Issues related to documentation are well known and addressed in a follow-up ticket. Issue 4 will be addressed when revisiting all n4js-libs. Issue 6 will be covered by the new VSCode functionality. Other issues are solved (see above).
